### PR TITLE
nrf53: fix RTC pretick power usage for events on RTC0

### LINKED ISSUE #63789: Increased power usage on nRF5340 with pretick
**Describe the bug**
After the PR https://github.com/zephyrproject-rtos/zephyr/pull/63148 was merged it was observed that 
in some Bluetooth scenarios the power usage on nRF5340 was significantly higher than expected.

**To Reproduce**
The issue was reproduced on internal test applications at Nordic Semiconductor using  RTC0. The Bluetooth was advertising with period around 100ms. In the sleep periods between radio activity the device consumed around 63uA average current instead of around 3uA.

**Expected behavior**
The nRF5340 in sleep periods between Bluetooth advertisements should show no increase in power consumption.

**Impact**
Quick discharge of battery.

### PR DESCRIPTION
**Pull request by ankuns**
Fixes #63789

For RTC0 events the RTC1 pretick event was not cleared what caused the WDT to be not stopped. This resulted in increased power usage.

### CODE REVIEW

File: soc/arm/nordic_nrf/nrf53/soc.c [hunk 1]
```diff
@@ -415,6 +415,10 @@ static void rtc_pretick_rtc_isr_hook(void)
 void rtc_pretick_rtc0_isr_hook(void)
 {
 	rtc_pretick_rtc_isr_hook();
+
+	if (IS_ENABLED(CONFIG_SOC_NRF53_RTC_PRETICK)) {
```
Comment (anangl) [inline soc/arm/nordic_nrf/nrf53/soc.c:419]: This `if` is redundant as the whole thing is inside https://github.com/zephyrproject-rtos/zephyr/blob/bf4652e842c56737b66fe4d50b4dcef41010a3bb/soc/arm/nordic_nrf/nrf53/soc.c#L377

And why not do this event clearing in `rtc_pretick_rtc_isr_hook` (it is common for RTC0 and RTC1)?

**Review by anangl — DISMISSED** [review]

File: soc/arm/nordic_nrf/nrf53/soc.c (hunk 1 above)
Comment (ankuns) [reply soc/arm/nordic_nrf/nrf53/soc.c]: fixed

**Comment by ankuns** [comment]
Pushed additional commit, marking DNM for a while

**Review by anangl — APPROVED** [review]

**Comment by jhedberg** [comment]
@ankuns @anangl there doesn't seem to be any "Fixes #12345" entry. Please add one (required at this stage of the release process if you want this for 3.5)

**Review by gmarull — APPROVED** [review]

**Comment by ankuns** [comment]
@jhedberg edited description, added "Fixes #63789"

**Pull request merged**